mixer/adapter/solarwinds/internal/appoptics/batching_test.go

```diff
@@ -37,32 +38,24 @@ func TestBatchMeasurements(t *testing.T) {
 
 		batchSize := 100
 
-		finish := make(chan bool)
-
-		go func() {
-			BatchMeasurements(prepChan, pushChan, stopChan, batchSize, logger)
-			finish <- true
-		}()
-
-		go func() {
-			measurements := make([]*Measurement, 0)
-			for i := 0; i < batchSize+1; i++ {
-				measurements = append(measurements, &Measurement{})
+		go BatchMeasurements(prepChan, pushChan, stopChan, batchSize, logger)
+
+		measurements := make([]*Measurement, 0)
+		for i := 0; i < batchSize+1; i++ {
+			measurements = append(measurements, &Measurement{})
+		}
+		prepChan <- measurements
+		count := 0
+		for range pushChan {
+			count++
+			if count == 1 {
+				break
 			}
-			prepChan <- measurements
-			stopChan <- struct{}{}
-			<-finish
-			close(prepChan)
-			close(pushChan)
-			count := 0
-			for range pushChan {
-				count++
-			}
-			if count != 1 {
-				t.Errorf("Batching is not working properly. Expected batches is 1 but got %d", count)
```

Comment:
curious: are we still testing this condition? It seems like we no longer are validating the size of the batching.

Reply:
It will fail and timeout if the it didn't get the push channel. it expect a push because the metric batch is larger than the batch size. It retains the original behavior.

Reply:
If we are going to rely on timeouts, can we use a `for` loop that selects between that and a restricted timeout?  Waiting on test timeouts seems non-ideal.

Is the case that this will always send 0 or 1?  Before, it seems possible to have more than 1 batch (also an error condition). Is that check no longer needed?

Reply:
+1. It looks like this won't cover the more-than-one-batch case anymore.

Also, I am not sure about the behavior of the batching code either. If the shutdown channel is tickled, it will drop the buffered events on the floor. I am not familiar with the rest of the code, but that seems problematic.